src/integration.test.ts

```diff
@@ -58,65 +93,93 @@ describe("Integration Test", (): void => {
     });
   });
 
-  const mockedExec = async (load: boolean, command: string): Promise<void> => {
-    const stdout = "standard output from Bash command";
-    const stderr = "error output from Bash command";
-    child_process.exec.mockImplementationOnce(
-      (_command: any, _options: any, callback: any): any => {
-        callback(null, { stdout, stderr });
-      }
-    );
-
-    await (load ? docker.loadDockerImages() : docker.saveDockerImages());
-
-    expect(core.getInput).nthCalledWith(1, "key", { required: true });
-    expect(core.info).nthCalledWith<[string]>(1, command);
-    expect(child_process.exec).lastCalledWith(
+  const assertExecBashCommand = (n: number, command: string): void => {
+    const infoN = 2 * n - 1; // core.info is called twice for each time docker image list is called.
+    expect(core.info).nthCalledWith<[string]>(infoN, command);
+    expect(child_process.exec).nthCalledWith(
+      n,
       command,
       EXEC_OPTIONS,
       expect.anything()
     );
-    expect(core.info).lastCalledWith(stdout);
-    expect(core.error).lastCalledWith(stderr);
+    expect(core.error).lastCalledWith(STDERR);
     expect(core.setFailed).not.toHaveBeenCalled();
   };
 
-  test("cache misses, then hits", async (): Promise<void> => {
-    // Attempt first cache restore.
-    await docker.loadDockerImages();
-
-    // Expect cache miss since cache has never been saved.
+  const assertListOrLoad = (command: string): void => {
     expect(core.getInput).nthCalledWith(1, "key", { required: true });
+    assertExecBashCommand(1, command);
+  };
+
+  const assertLoadCacheMiss = (): void => {
+    expect(core.saveState).lastCalledWith(
+      docker.DOCKER_IMAGES_LIST,
+      "test-docker-image:v1"
+    );
+    expect(core.info).lastCalledWith(dockerImages.join("\n"));
```

Comment:
This is duplicitous with the assert currently in the mock implementation of `util.exec`.